cg/meta/upload/gt.py

```diff
@@ -43,93 +47,142 @@ def data(self, analysis: Analysis) -> dict:
         """
         case_id = analysis.case.internal_id
         LOG.info(f"Fetching upload genotype data for {case_id}")
-        hk_version = self.hk.last_version(case_id)
-        hk_bcf = self.get_bcf_file(hk_version)
-        data = {"bcf": hk_bcf.full_path}
+        hk_bcf = self._get_bcf_file(case_id=case_id)
+        gt_data: dict = {"bcf": hk_bcf.full_path}
         if analysis.workflow in [Workflow.BALSAMIC, Workflow.BALSAMIC_UMI]:
-            data["samples_sex"] = self._get_samples_sex_balsamic(case_obj=analysis.case)
+            gt_data["samples_sex"] = self._get_samples_sex_balsamic(case=analysis.case)
         elif analysis.workflow == Workflow.MIP_DNA:
-            data["samples_sex"] = self._get_samples_sex_mip(
-                case_obj=analysis.case, hk_version=hk_version
-            )
+            gt_data["samples_sex"] = self._get_samples_sex_mip_dna(case=analysis.case)
+        elif analysis.workflow == Workflow.RAREDISEASE:
+            gt_data["samples_sex"] = self._get_samples_sex_raredisease(case=analysis.case)
         else:
             raise ValueError(f"Workflow {analysis.workflow} does not support Genotype upload")
-        return data
-
-    def _get_samples_sex_mip(self, case_obj: Case, hk_version: Version) -> dict:
-        qc_metrics_file = self.get_qcmetrics_file(hk_version)
-        analysis_sexes = self.analysis_sex(qc_metrics_file)
-        samples_sex = {}
-        for link_obj in case_obj.links:
-            sample_id = link_obj.sample.internal_id
+        return gt_data
+
+    def upload(self, data: dict, replace: bool = False):
+        """Upload data about genotypes for a family of samples."""
+        self.gt.upload(str(data["bcf"]), data["samples_sex"], force=replace)
+
+    @staticmethod
+    def _is_suitable_for_genotype_upload(case: Case) -> bool:
+        """Check if a cancer case is contains WGS and normal sample."""
```

Comment:
```suggestion
        """Check if a cancer case is contains WGS and normal samples."""
```
What is normal here?